File: backend/models/item.go
package models

import (
	"go.mongodb.org/mongo-driver/bson/primitive"
	"time"
)

type Item struct {
	ID           primitive.ObjectID `bson:"_id,omitempty" json:"id"`
	Tipo         string             `bson:"tipo" json:"tipo"`
	Title        string             `bson:"title" json:"title"`
	Descripcion  string             `bson:"descripcion" json:"descripcion"`
	Estado       string             `bson:"estado" json:"estado"`
	Cantidad     int                `bson:"cantidad" json:"cantidad"`
	Imagen       string             `bson:"imagen,omitempty" json:"imagen,omitempty"`
	Imagenes     []string           `bson:"imagenes,omitempty" json:"imagenes,omitempty"`
	ArrendadoPor string             `bson:"arrendadoPor" json:"arrendadoPor"`
	CreatedAt    time.Time          `bson:"createdAt" json:"createdAt"`
	UpdatedAt    time.Time          `bson:"updatedAt" json:"updatedAt"`
}

File: src/models/Item.js
import mongoose from 'mongoose';

// Modelo único de Item. Todas las rutas deben importarlo desde aquí:
// definir el schema en cada ruta hace que gane el primero en registrarse
// y que los campos nuevos se pierdan silenciosamente (strict mode).
const ItemSchema = new mongoose.Schema({
  tipo: { type: String, required: true },
  title: { type: String, required: true },
  descripcion: { type: String, required: true },
  estado: { type: String, required: true },
  cantidad: { type: Number, default: 1, required: true },
  imagenes: [{ type: String }],
  arrendadoPor: { type: String, default: "NaN" },
  vendidoA: { type: String, default: "NaN" },
  accion: { type: String, default: "agregado" },
}, {
  timestamps: true
});


const Item = mongoose.models.Item || mongoose.model('Item', ItemSchema);

export default Item;
